addons/sale_management/models/sale_order_line.py

```diff
@@ -19,7 +19,9 @@ def _compute_name(self):
                 for template_line in line.order_id.sale_order_template_id.sale_order_template_line_ids:
                     if line.product_id == template_line.product_id:
```

Comment:
```suggestion
                    if line.product_id == template_line.product_id and template_line.name:
                        # If a specific description was set on the template, use it
                        # Otherwise the description is handled by the super call
```